internal/user.go

```diff
@@ -0,0 +1,171 @@
+package internal
```

Comment:
won't there types get published if they are not in an _test file?

Reply:
`internal` is a special package in Go. It only exposes its content to the parent package.

Reply:
thanks! I didn't know that. Resolving.